File: Fragment-Demo/app/src/main/java/com/hasib/fragmentdemo/MainActivity.java
package com.hasib.fragmentdemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        addFragment();
    }

    private void addFragment() {
        FragmentManager manager = getSupportFragmentManager();
        FragmentTransaction transaction = manager.beginTransaction();

        SampleFragment sampleFragment = new SampleFragment();

        transaction.add(R.id.fragmentContainer, sampleFragment);
        transaction.addToBackStack("fragmentStack1");
        transaction.commit();
    }
}

File: _Kotlin-Camouflage/AndroidHiltDemo/app/src/main/java/com/hasib/androidhiltdemo/ui/MainActivity.kt
package com.hasib.androidhiltdemo.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.hasib.androidhiltdemo.R
import com.hasib.androidhiltdemo.navigator.AppNavigator
import com.hasib.androidhiltdemo.navigator.Screens
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

/**
 * Main activity of the application.
 *
 * Container for the Buttons & Logs fragments. This activity simply tracks clicks on buttons.
 */
@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var navigator: AppNavigator

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        savedInstanceState ?: navigator.navigateTo(Screens.BUTTONS)
    }

    override fun onBackPressed() {
        super.onBackPressed()

        if (supportFragmentManager.backStackEntryCount == 0) {
            finish()
        }
    }
}